### PR TITLE
test: expand mutation testing across workspaces

### PR DESCRIPTION
**Pull request by hiroppy**
# Summary

- Run Stryker mutation testing across all six workspaces that have unit tests: crawler, web, analytics, date-utils, db, and meta.
- Orchestrate the workspace gates through `pnpm test:mutation` and Turborepo, with shared catalog-pinned Stryker dependencies.
- Keep browser/E2E/Storybook concerns out of mutation runs and enforce regression thresholds derived from each workspace's measured baseline.

## Linear Issue

- [HIR-113](https://linear.app/hiroppy/issue/HIR-113/mutation-testingを実施する)

### Selected Quality Characteristics

| Quality characteristic | Why it is relevant | Acceptance criteria |
| ---------------------- | ------------------ | ------------------- |
| Functional suitability | Passing unit tests can still contain weak assertions around branches, boundaries, and error paths. | Production mutants are exercised by each workspace's unit suite and every configured baseline gate passes. |
| Maintainability | The quality signal must remain reproducible and locally attributable in a monorepo. | One root command runs explicit workspace-local targets, exclusions, and thresholds. |
| Performance efficiency | Mutation testing multiplies test execution and can overload CI when browser or computational hot paths are included. | Unit-only configs, 50% worker concurrency, and static-mutant exclusions for apps keep the full gate bounded. |

### Test Techniques

| Test technique | Selection rationale | Expected coverage |
| -------------- | ------------------- | ----------------- |
| Boundary value analysis | Date, amount, pagination, and simulation helpers contain numeric and calendar limits. | Lower/upper boundaries and invalid partitions are challenged by generated mutants. |
| Branch testing | Stryker mutates conditionals, operators, and default branches across tested production modules. | Surviving and uncovered mutants are reported per file and regression thresholds block material score loss. |
| Error guessing | Browser-context serialization and high-cost Monte Carlo paths are known mutation-run failure modes. | Dry-run failures are reproduced, classified, and isolated without weakening their normal unit tests. |

### Primary Test Conditions

- Run mutation testing for crawler, web, analytics, date-utils, db, and meta from the repository root.
- Use unit-only Vitest configurations for crawler and web; do not invoke crawler E2E or web Storybook projects.
- Preserve the existing date-utils 80% gate and enforce measured regression floors for the newly covered workspaces.
- Keep ordinary unit tests, type checking, lint, formatting, and unused-code checks green.

### Out-of-Scope Risks

- Workspaces without unit-test scripts (`mcp`, `simulator`, `db-e2e`) are not mutation targets.
- Crawler functions serialized into `page.evaluate` are excluded because Stryker's instrumentation state does not exist in the browser execution context.
- The web Monte Carlo module is excluded from mutation instrumentation because its integrated sensitivity test exceeds the dry-run timeout under instrumentation; its 64 normal unit tests remain in the regression suite.
- Seed data, schemas, type-only modules, and environment/bootstrap entry points without unit coverage are excluded rather than lowering scores with non-executable mutants.

## Verification

- [x] Mutation testing
- [x] Unit and Storybook tests
- [x] Type checking
- [x] Lint
- [x] Formatting
- [x] Unused-code analysis
- [ ] E2E tests

### Commands Executed

```text
pnpm test:mutation — 6/6 workspace tasks passed in 20m47s with TypeScript checker enabled\n  crawler 43.85% (break 40)\n  web 59.26% (break 55)\n  analytics 52.54% (break 45)\n  date-utils 94.85% (break 80)\n  db 46.59% (break 45)\n  meta 100% (break 80)
pnpm test — 1,297 tests passed, 17 skipped
pnpm turbo typecheck — 9/9 tasks passed
pnpm lint — passed
pnpm format:check — passed
pnpm knip — passed\nGitHub CI — 8/8 checks passed; test job includes the mutation gate and report artifact
```

### Checks Not Run

- E2E tests — test-tooling configuration only; no application runtime behavior changed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "db:clean": "rm -f data/moneyforward.db data/moneyforward.db-wal data/moneyforward.db-shm",
     "db:dev": "pnpm --filter @mf-dashboard/crawler dev:scrape",
     "test": "turbo test",
+    "test:mutation": "pnpm --filter @mf-dashboard/date-utils test:mutation",
```
[the remark this answers is not in the document]
Comment (hiroppy) [reply package.json]: 対応しました。`.github/workflows/ci.yml` の必須 test job に `pnpm test:mutation` を追加し、80% threshold を PR CI で常時検証します。ローカル再実行でも 138/149 killed、92.62% で gate 通過を確認済みです（7c4edb9）。

**Review by hiroppy — COMMENTED** [review]
全体には適応できない？

**Comment by hiroppy** [comment]
対応しました。`pnpm test:mutation` を Turbo 経由で、unit test を持つ crawler / web / analytics / date-utils / db / meta の6 workspace 全体へ展開しました（2b4176a）。各 workspace は実測 baseline に基づく regression gate を持ち、全体実行は6/6成功・約7分です。browser context に関数を渡す crawler の3 scraper と、instrumentation 下で dry run timeout になる web の Monte Carlo は技術的理由で mutation 対象外ですが、通常 unit test は引き続き全件実行しています。PR本文にも対象・閾値・除外理由を追記しました。

**Review by hiroppy — COMMENTED** [review]
@stryker-mutator/typescript-checkerを追加し、typescriptに対応

**Comment by hiroppy** [comment]
対応しました。`@stryker-mutator/typescript-checker` 9.6.1 を catalog と mutation testing 対象の6 workspaceに追加し、各 Stryker config で `checkers: ["typescript"]` と plugin を明示しました（pnpm では明示 plugin 読み込みが必要です）。`pnpm test:mutation` は6/6成功（20分47秒）で、各 workspace の checker process 起動と型不正 mutant の `CompileError` 分類を確認しています。通常 test / typecheck / lint / format / knip も成功しています。

File: apps/crawler/stryker.config.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "!src/index.ts"
+  ],
+  "packageManager": "pnpm",
+  "plugins": ["@stryker-mutator/typescript-checker", "@stryker-mutator/vitest-runner"],
+  "reporters": ["clear-text", "progress", "html", "json"],
+  "testRunner": "vitest",
+  "thresholds": {
+    "high": 80,
+    "low": 40,
+    "break": 40
```
[the remark this answers is not in the document]
Comment (hiroppy) [reply apps/crawler/stryker.config.json:22]: ご指摘の `break` と `high` の違いは認識していますが、今回は全 workspace に 80% を強制する意図ではありません。PR 本文の Primary Test Conditions にある通り、date-utils の既存 80% gate は維持し、新規対象は実測 baseline（crawler 43.85%、web 59.26%、analytics 52.54%、db 46.59%）を下回る regression を `break` で阻止します。`high: 80` は目標帯の表示で、現 baseline が 80% 未満の4 workspaceへ `break: 80` を設定すると、この PR の最新実測でも CI が常時失敗します。したがって workspace 別 `break` は意図した設定として維持します。
